File: Scripts/results/generated_code_results/Flutter/Social Networking/SocialMessagingApp/ContactsScreen.dart
import 'package:flutter/material.dart';
class ContactsScreen extends StatelessWidget {
final List<Map<String, dynamic>> contacts = [
{'name': 'Antonio Banderas', 'status': 'Online', 'initials': 'AB'},
{'name': 'Billie Eilish', 'status': 'Online', 'initials': 'BE'},
{'name': 'Chris Evans', 'status': 'Last seen today at 8:40', 'initials': 'CE'},
{'name': 'Dasha Taran', 'initials': 'DT'},
{'name': 'Dwayne Johnson', 'status': 'Last seen today at 6:02', 'initials': 'DJ'},
{'name': 'Emily Blunt', 'status': 'Last seen today at 8:33', 'initials': 'EB'},
];
@override
Widget build(BuildContext context) {
return Scaffold(
appBar: AppBar(
title: Text('Contacts'),
elevation: 0,
),
body: Column(
children: [
ListTile(
leading: Icon(Icons.person_add),
title: Text('Invite friends'),
onTap: () {},
),
ListTile(
leading: Icon(Icons.location_on),
title: Text('Find people nearby'),
onTap: () {},
),
Expanded(
child: ListView.builder(
itemCount: contacts.length,
itemBuilder: (context, index) {
final contact = contacts[index];
return ListTile(
leading: CircleAvatar(
backgroundColor: Colors.grey[300],
child: Text(contact['initials']),
),
title: Text(contact['name']),
subtitle: Text(contact['status'] ?? ''),
onTap: () {
Navigator.pushNamed(context, 'Chat', arguments: contact);
},
);
},
),
),
],
),
bottomNavigationBar: BottomNavigationBar(
currentIndex: 1,
type: BottomNavigationBarType.fixed,
items: [
BottomNavigationBarItem(
icon: Icon(Icons.message),
label: 'Messages',
),
BottomNavigationBarItem(
icon: Icon(Icons.people),
label: 'Contacts',
),
BottomNavigationBarItem(
icon: Icon(Icons.phone),
label: 'Calls',
),
BottomNavigationBarItem(
icon: Icon(Icons.person),
label: 'Profile',
),
],
onTap: (index) {
if (index == 0) {
Navigator.pushNamed(context, 'Messages');
}
},
),
);
}
}

File: Scripts/results/generated_code_results/RN_gpt/Social Networking/SocialMessagingApp/ContactsScreen.js
import React from 'react';
import { View, Text, TouchableOpacity, Image, ImageBackground, StyleSheet, ScrollView } from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';

const ContactsScreen = ({ navigation }) => {
  return (
    <View style={styles.container}>
      <Text style={styles.headerText}>Contacts</Text>
      <ScrollView>
        <View style={styles.actionContainer}>
          <TouchableOpacity style={styles.actionButton}>
            <Icon name="person-add" size={20} color="#000" />
            <Text style={styles.actionText}>Invite friends</Text>
          </TouchableOpacity>
          <TouchableOpacity style={styles.actionButton}>
            <Icon name="location-on" size={20} color="#000" />
            <Text style={styles.actionText}>Find people nearby</Text>
          </TouchableOpacity>
        </View>

        <View style={styles.contactList}>
          {contactsData.map((contact, index) => (
            <TouchableOpacity 
              key={index} 
              style={styles.contactItem} 
              onPress={() => navigation.navigate('Chat', { contactName: contact.name })}
            >
              <ImageBackground 
                source={require('../assets/snack-icon.png')} 
                style={styles.avatar} 
              >
                <Text style={styles.avatarText}>{contact.initials}</Text>
              </ImageBackground>
              <View style={styles.contactDetails}>
                <Text style={styles.contactName}>{contact.name}</Text>
                <Text style={styles.contactStatus}>{contact.status}</Text>
              </View>
            </TouchableOpacity>
          ))}
        </View>
      </ScrollView>

      <View style={styles.bottomNavigation}>
        <TouchableOpacity onPress={() => navigation.navigate('Messages')}>
          <Text style={styles.navText}>Messages</Text>
        </TouchableOpacity>

        <TouchableOpacity>
          <Text style={styles.navTextActive}>Contacts</Text>
        </TouchableOpacity>

        <TouchableOpacity onPress={() => navigation.navigate('Calls')}>
          <Text style={styles.navText}>Calls</Text>
        </TouchableOpacity>

        <TouchableOpacity onPress={() => navigation.navigate('Profile')}>
          <Text style={styles.navText}>Profile</Text>
        </TouchableOpacity>
      </View>
    </View>
  );
};

const contactsData = [
  { name: 'Antonio Banderas', status: 'Online', initials: 'AB' },
  { name: 'Billie Eilish', status: 'Online', initials: 'BE' },
  { name: 'Chris Evans', status: 'Last seen today at 8:40', initials: 'CE' },
  { name: 'Dasha Taran', status: '', initials: 'DT' },
  { name: 'Dwayne Johnson', status: 'Last seen today at 6:02', initials: 'DJ' },
  { name: 'Emily Blunt', status: 'Last seen today at 8:33', initials: 'EB' },
];

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#FFFFFF',
  },
  headerText: {
    fontSize: 24,
    fontWeight: 'bold',
    padding: 20,
  },
  actionContainer: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    marginBottom: 10,
  },
  actionButton: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  actionText: {
    fontSize: 16,
    marginLeft: 5,
  },
  contactList: {
    paddingHorizontal: 20,
  },
  contactItem: {
    flexDirection: 'row',
    alignItems: 'center',
    marginVertical: 10,
    paddingVertical: 10,
    borderBottomWidth: 1,
    borderBottomColor: '#DDD',
  },
  avatar: {
    width: 50,
    height: 50,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#EEE',
    borderRadius: 25,
    marginRight: 15,
  },
  avatarText: {
    fontSize: 18,
    fontWeight: 'bold',
  },
  contactDetails: {
    flex: 1,
  },
  contactName: {
    fontSize: 18,
    fontWeight: '600',
  },
  contactStatus: {
    color: 'gray',
  },
  bottomNavigation: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    paddingVertical: 20,
    borderTopWidth: 1,
    borderTopColor: '#DDD',
  },
  navText: {
    fontSize: 16,
    color: 'gray',
  },
  navTextActive: {
    fontSize: 16,
    fontWeight: 'bold',
    color: '#000',
  },
});

export default ContactsScreen;